src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -767,23 +760,47 @@ private[snowflake] class SnowflakeSQLStatement(
       case (element, index) =>
         element match {
           case ele: StringVariable =>
-            statement.setString(index + 1, ele.variable)
+            if (ele.variable.isDefined)
```

Comment:
Since NULL is handled in executeWithBindVariable(), I think there may be no test coverage for them.
I suggest to add some unit test for it. Actually, we can add unit test in IT, so we can have a real connection to make the unit test easier. I have some test for stage in StageSuite.scala. FYI.

Reply:
UnitTestWithConnectionSuite should be a better reference.